src/sql/inspect.py

```diff
@@ -0,0 +1,84 @@
+from sqlalchemy import inspect
+from prettytable import PrettyTable
+
+
+from sql.connection import Connection
+from sql.telemetry import telemetry
+
+
+def _get_inspector(conn):
+    if conn:
+        return inspect(conn)
+
+    if not Connection.current:
+        raise RuntimeError("No active connection")
+    else:
+        return inspect(Connection.current.session)
+
+
+class DatabaseInspection:
+    def __repr__(self) -> str:
+        return self._table_txt
+
+    def _repr_html_(self) -> str:
+        return self._table_html
+
+
+class Tables(DatabaseInspection):
+    """
+    Displays the tables in a database
+    """
+
+    def __init__(self, conn=None) -> None:
+        inspector = _get_inspector(conn)
+
+        self._table = PrettyTable()
+        self._table.field_names = ["Name"]
+
+        for row in inspector.get_table_names():
+            self._table.add_row([row])
+
+        self._table_html = self._table.get_html_string()
+        self._table_txt = self._table.get_string()
+
+
+class Columns(DatabaseInspection):
+    """
+    Represents the columns in a database table
+    """
+
+    def __init__(self, name, schema, conn=None) -> None:
+        inspector = _get_inspector(conn)
+
+        columns = inspector.get_columns(name, schema)
+
+        if not columns:
+            if schema:
+                raise ValueError(
+                    f"There is no table with name {name!r} in schema {schema!r}"
+                )
+            else:
+                raise ValueError(
+                    f"There is no table with name {name!r} in the default schema"
+                )
+
+        self._table = PrettyTable()
+        self._table.field_names = list(columns[0].keys())
+
+        for row in columns:
+            self._table.add_row(list(row.values()))
+
+        self._table_html = self._table.get_html_string()
+        self._table_txt = self._table.get_string()
+
+
+@telemetry.log_call()
+def get_table_names():
```

Comment:
good point. I think we should fix this in the core package. (do a `replace("_", "-")` so we don't have to do it here to keep it simple. @idomic thoughts?

Reply:
Yeah we can have a `.replace()` on the string, an easy fix.
I think get_table_names though might be too much? Why not just pass it tables? Do we have other functions that relates to tables?

Reply:
Ensure we're on the same page: the `.replace` will go to the `core` package, affecting this and other events.

> I think get_table_names though might be too much? Why not just pass it tables? Do we have other functions that relates to tables?

We should keep event_name=function_name; this will allow us to identify where each event is coming from.

Reply:
Sure